### PR TITLE
Make Custom Config Optional

### PR DESCRIPTION
**Pull request by ashishjayamohan**
- Made custom config in tableConfig optional
- Changed key for custom config from "metadata" to "customConfig"
- Added nullity check for getter

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
Hmm.. I think this change is backward incompatible. But I'm +100 on making it optional.
@jackjlli do you use this field internally?

**Comment by jackjlli** [comment]
> Hmm.. I think this change is backward incompatible. But I'm +100 on making it optional. @jackjlli do you use this field internally?

Yes, this field is actively being used internally, so it's a `required` field for LinkedIn use cases. But it should make no harm to make it optional. 
@ashishjayamohan what's the rational of making it optional? If there is nothing specified in it, it shouldn't occupy too much space though.

**Comment by ashishjayamohan** [comment]
Honestly, the rationale is pretty simplistic - it just seems like excess overhead if not needed. Additionally, making it optional (and please correct me on this if I'm wrong) should not affect the current invocations even if it is a required field for some use cases. This is also my first PR here, so please let me know if there's something I'm missing - I'd love to learn more!

**Comment by Jackie-Jiang** [comment]
The main concern is the change of the key from "metadata" to "customConfig". This can prevent old table config being properly loaded, and is the reason why all backward compatibility tests are failing

**Comment by jackjlli** [comment]
We couldn't afford to rename the `metadata` field to `customConfig` as that will break all the existing logic regardless of internal one and OSS. I'd prefer just keeping the field name to remain unchanged and just mark the custom config optional.

**Comment by ashishjayamohan** [comment]
That makes sense. I've switched the custom config key back to "metadata."

File: pinot-core/src/test/java/org/apache/pinot/queries/ForwardIndexDisabledMultiValueQueriesWithReloadTest.java [hunk 1]
```diff
@@ -63,8 +63,8 @@
  * TODO: Find a good way to consolidate this with ForwardIndexDisabledMultiValueQueriesTest
  * The <code>ForwardIndexDisabledMultiValueQueriesWithReloadTest</code> class sets up the index segment for the
  * no forward index multi-value queries test with reload.
- * <p>There are totally 14 columns, 100000 records inside the original Avro file where 10 columns are selected to build
- * the index segment. Selected columns information are as following:
+ * <p>There are totally 14 columns, 100000 records inside the original Avro file where 10 columns are selected to
+ * build the index segment. Selected columns information are as following:
```
Comment (Jackie-Jiang) [inline pinot-core/src/test/java/org/apache/pinot/queries/ForwardIndexDisabledMultiValueQueriesWithReloadTest.java:67]: This change is unrelated

File: pinot-spi/src/main/java/org/apache/pinot/spi/config/table/TableConfig.java [hunk 2]
```diff
@@ -77,14 +77,13 @@ public class TableConfig extends BaseJsonConfig {
...
 
-  // TODO: Make TableCustomConfig optional and use another key other than 'metadata'
-  private TableCustomConfig _customConfig;
-
   /* OPTIONAL FIELDS */
 
   @JsonPropertyDescription("Resource quota associated with this table")
   private QuotaConfig _quotaConfig;
 
+  private TableCustomConfig _customConfig;
```
Comment (Jackie-Jiang) [inline pinot-spi/src/main/java/org/apache/pinot/spi/config/table/TableConfig.java:85]: (minor) Put this above `QuotaConfig` to maintain the same order

**Review by Jackie-Jiang — COMMENTED** [review]
There is an extra unrelated binary file `pinot-plugins.tar.gz` included. Can you please remove it?

**Comment by ashishjayamohan** [comment]
@Jackie-Jiang Yeah, I'm unsure of how that got in there. I don't think I modified anything in that directory to begin with, but I removed it.

**Review by Jackie-Jiang — APPROVED** [review]

**Review by jackjlli — APPROVED** [review]
LGTM. Thanks for making the changes! 👍
